### PR TITLE
For #12298:  App should not be locked in landscape when a tab or custom tab loads while in pip mode

### PR DESCRIPTION
**Pull request by indurs**
App should not be locked in landscape while in pip mode when a tab or custom tab loads

This allows a tab or custom tab to be loaded in device's current orientation while pip mode is active

Co-Authored-By: Mugurell <Mugurell@users.noreply.github.com>

https://user-images.githubusercontent.com/12825812/172154727-29e46490-d5e6-46ac-a3bc-cbd74cb14a92.mp4

@Mugurell, this check will ensure that the orientation will be applied only for the tab or the custom tab that have the corresponding active media state which is being filtered in the flow. 
   https://github.com/indurs/android-components/blob/issuefix25413/components/feature/media/src/main/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeature.kt#L53

We need to pass the customTabSessionId (if there is one otherwise null) to MediaSessionFullscreenFeature.  BaseBrowserFragment in Fenix and Focus has to be updated to pass the customTabSessionId  to MediaSessionFullscreenFeature.

Please let me know if I've missed anything.

I've also tested for the following scenarios while in pip mode
Search widget, Home Shortcut, Focus share, links

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/media/src/test/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeatureTest.kt [hunk 1]
```diff
@@ -94,7 +101,7 @@ class MediaSessionFullscreenFeatureTest {
 
     @Test
     fun `GIVEN the currently selected tab plays landscape media WHEN it enters fullscreen THEN set orientation to landscape`() {
-        val activity: Activity = mock()
+        val activity = Robolectric.buildActivity(Activity::class.java).setup().get()
```
Comment (Mugurell) [inline components/feature/media/src/test/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeatureTest.kt:104]: Not sure why this change is needed.
Verifying that `setRequestedOrientation` is called or not on the mocked activity should be enough.

File: components/feature/media/src/test/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeatureTest.kt [hunk 2]
```diff
@@ -254,4 +265,59 @@ class MediaSessionFullscreenFeatureTest {
 
         assertEquals(ActivityInfo.SCREEN_ORIENTATION_USER, activity.requestedOrientation)
     }
+
+    @Suppress("Deprecation")
+    @Test
+    @Config(sdk = [Build.VERSION_CODES.N])
+    fun `GIVEN the currently selected tab is in pip mode WHEN a custom tab loads THEN display custom tab in device's current orientation`() {
+        val activity = Robolectric.buildActivity(Activity::class.java).setup().get()
```
Comment (Mugurell) [inline components/feature/media/src/test/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeatureTest.kt:273]: Same as above, maybe we don't need a real instance of an activity?

**Review by Mugurell — COMMENTED** [review]
Looks good.
Tested on Fenix and the issue reported there is resolved.
Just need this patch to reference an A-C ticket (I've created https://github.com/mozilla-mobile/android-components/issues/12298 for this) in the commit title and in a changelog entry.
Also the changes to use a real instance of an activity doesn't seem to be needed and we can avoid a tighter coupling to Robolectric?

Since this is a breaking change we will need accompanying patches on both Fenix and Focus (for which I've created https://github.com/mozilla-mobile/focus-android/issues/7190 to address the bug) to be ready for merging before actually landing this patch.

File: components/feature/media/src/test/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeatureTest.kt (hunk 1 above)
Comment (indurs) [reply components/feature/media/src/test/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeatureTest.kt:104]: You're right mock and verify is enough. Will update

File: components/feature/media/src/test/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeatureTest.kt (hunk 2 above)
Comment (indurs) [reply components/feature/media/src/test/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeatureTest.kt:273]: @Mugurell , I'm slowly becoming familiar with mock and RoboElectric. Just using mock is not working here probably it's just a stub. It seems that an activity object is needed to set the pip mode for the activity. Also MediaSessionFullScreenFeature needs to check the activity's orientation after pip mode is active.
Please correct me if I'm wrong

Comment (indurs) [reply components/feature/media/src/test/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeatureTest.kt:273]: Thank You ! Will try the above method next time

**Review by Mugurell — APPROVED** [review]
Changes look good, PRs for addressing these already exist for Fenix and Focus, looks ready to land.

File: docs/changelog.md [hunk 3]
```diff
@@ -20,6 +20,9 @@ permalink: /changelog/
 * **feature-prompts**:
   * Added optional `addressPickerView` and `onManageAddresses` parameters through `AddressDelegate` to `PromptFeature` for a new `AddressPicker` to display a view for selecting addresses to autofill into a site. [#12061](https://github.com/mozilla-mobile/android-components/issues/12061)
 
+* **feature-media**
```
Comment (Mugurell) [inline docs/changelog.md:23]: One small change needed here: New changelog entries are to be added at the top for this file to later have changes in chronological order.
Will address this in the PR separate PR for landing this.

Comment (indurs) [reply docs/changelog.md]: Sure.
